### PR TITLE
Convert `usage-data` store to an options Pinia store

### LINKED ISSUE #1026: Convert `usage-data` store from Vuex to Pinia
## Problem

As part of the conversion of the Vuex store modules to Pinia, we need to convert the [usage-data store](https://github.com/WordPress/openverse-frontend/blob/e0ac241dfe8b2b4b7eaff8eaee7e06eda0dce197/src/store/user.js) to Pinia. This store uses the `usageDataService` which needs to be passed as a parameter to the function that creates a store definition function, something like:
```js
import UsageDataService from '...'
const createUseUsageDataStore = (usageDataService=UsageDataService) => defineStore('usage-data', () => {})
export const useUsageDataStore = createUseUsageDataStore(UsageDataService)
```

## Description

This is the general plan for conversion, not all steps might be necessary:
- use defineStore to create the store.
-  convert state to reactive
- convert getters without params to computed and getters with parameters to functions.
- convert actions and mutations to functions. They will no longer use the context or state parameter, and will, instead of using an all-capital constant for a name, be renamed to use camelCase (e.g. [FETCH_MEDIA] will become fetchMedia).
- return the necessary state properties, getters and actions.
-  remove the unused constants from - ~/constants/mutation-types.js, ~/constants/action-types.js, ~/constants/store-modules.js.
- refactor the use of the store in components.
-  add or refactor the store unit test.
- refactor the unit tests for components that use this store.
- add e2e tests for the changes in components.

## Additional context

The RFC for Pinia migration: https://github.com/WordPress/openverse/pull/174

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1026 by @obulat

## Description

This PR converts `usage-data` store to Pinia. This store handles sending of the analytics data points to the API, and doesn't contain any state, so it might be a good idea to refactor it into a composable in the future.
For now, I converted it from Vuex to Pinia so that we can keep converting other stores that depend on it. 

This store needs to access `$sentryReady` from the Nuxt context. Because the Pinia options store has  the context available on `this.$nuxt`, I used an `options` store (with actions, similar to Vuex), and not the `setup` store (similar to composable functions).

## Testing Instructions

Enable Sentry and run the app: `ENABLE_INTERNAL_ANALYTICS=true pnpm run dev`

Revert the fix for analytics endpoint API bug that results in double slashes at the end:
https://github.com/WordPress/openverse-frontend/blob/7d886d34eb84a76cf8e8c9c58e0b319b6dc0b348/src/data/usage-data-service.js#L17
This will cause analytics requests to throw a network error. When searching, opening the detail page, clicking on the 'Copy attribution', you should see the sentry post requests in the console. You will have to disable ad blocking extensions if you have any.
The unit tests have almost 100% coverage (the internal of Sentry are not covered).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/ImageDetails/ReuseSurvey.vue [hunk 1]
```diff
@@ -29,6 +27,18 @@ export default {
...
   components: { VLink },
   props: ['image'],
+  setup(props) {
+    const usageDataStore = useUsageDataStore()
+    const onReuseSurveyClick = async () => {
+      await usageDataStore.sendDetailPageEvent({
+        eventType: DETAIL_PAGE_EVENTS.REUSE_SURVEY,
+        resultUuid: props.image.id,
+      })
+    }
```
Comment (sarayourfriend) [inline src/components/ImageDetails/ReuseSurvey.vue:37]: Analytics requests shouldn't block; they don't have any bearing on the UI and blocking on them will make the UI less responsive.

```suggestion
    const onReuseSurveyClick = () => {
      usageDataStore.sendDetailPageEvent({
        eventType: DETAIL_PAGE_EVENTS.REUSE_SURVEY,
        resultUuid: props.image.id,
      })
    }
```

File: src/stores/usage-data.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          scope.setTag('event_name', eventName)
+          scope.setTag('request_url', error.config.url)
+        })
+      )
+  )
+
+export const createUseUsageDataStore = (usageDataService = UsageDataService) =>
+  defineStore('usage-data', {
+    actions: {
+      async sendSearchQueryEvent(params) {
```
Comment (sarayourfriend) [inline src/stores/usage-data.js:39]: We can unable these as `async`, they don't actually return `Promise`s and there's nothing to await in or outside of them.

File: src/components/ImageDetails/ReuseSurvey.vue (hunk 1 above)
Comment (zackkrida) [reply src/components/ImageDetails/ReuseSurvey.vue:37]: We should add a comment about this so folks don't forget in the future.

**Comment by obulat** [comment]
Closing this PR as it's replaced with #1084.

**Pull request closed without merging**